framework/src/node/consensus/certificate_generation/commit_pool.ts

```diff
@@ -169,15 +171,61 @@ export class CommitPool {
 		);
 	}
 	// TODO: To be updated in the issue https://github.com/LiskHQ/lisk-sdk/issues/6846
-	public getAggregageCommit(): AggregateCommit {
+	public async getAggregageCommit(apiContext: APIContext): Promise<AggregateCommit> {
 		const singleCommits = this._selectAggregateCommit();
 
-		return this._aggregateSingleCommits((singleCommits as unknown) as SingleCommit[]);
+		return this.aggregateSingleCommits(apiContext, (singleCommits as unknown) as SingleCommit[]);
 	}
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	private _aggregateSingleCommits(_singleCommits: SingleCommit[]): AggregateCommit {
-		return {} as AggregateCommit;
+
+	public async aggregateSingleCommits(
+		apiContext: APIContext,
+		singleCommits: SingleCommit[],
+	): Promise<AggregateCommit> {
+		if (singleCommits.length === 0) {
+			throw new Error('No single commit found');
+		}
+
+		const { height } = singleCommits[0];
+		// assuming this list of validators includes all validators corresponding to each singleCommit.validatorAddress
+		const { validators } = await this._bftAPI.getBFTParameters(apiContext, height);
+		const addressToBlsKey: dataStructures.BufferMap<Buffer> = new dataStructures.BufferMap();
+		const validatorKeys: Buffer[] = [];
+
+		for (const validator of validators) {
+			const validatorAccount = await this._validatorsAPI.getValidatorAccount(
+				apiContext,
+				validator.address,
+			);
+			addressToBlsKey.set(validator.address, validatorAccount.blsKey);
+			validatorKeys.push(validatorAccount.blsKey);
+		}
+
+		validatorKeys.sort((blsKeyA, blsKeyB) => blsKeyA.compare(blsKeyB));
+
+		interface PkSigPair {
+			publicKey: Buffer;
+			signature: Buffer;
+		}
+		const pubKeySignaturePairs: PkSigPair[] = [];
+
+		for (const commit of singleCommits) {
+			const publicKey = addressToBlsKey.get(commit.validatorAddress)!;
```

Comment:
instead of ignoring the undefined here, maybe it's good to check and throw if undefined?